lib/cocoapods/sandbox.rb

```diff
@@ -271,8 +271,8 @@ def store_podspec(name, podspec, _external_source = false, json = false)
         raise ArgumentError, "Unknown type for podspec: #{podspec.inspect}"
       end
 
-      spec ||= Specification.from_file(output_path)
-      spec.defined_in_file ||= output_path
+      spec = spec ? spec.dup : Specification.from_file(output_path)
```

Comment:
should this go into the `when` cases that require it? I noticed that the `when Specification` case already calls `dup`, so this would dup it twice. is this only needed in `when Pathname`?

Reply:
also probably worth a comment explaining why its necessary

Reply:
or remove the `.dup` above?

Reply:
agree on comment